app/code/core/Mage/Core/Model/Layout/Validator.php

```diff
@@ -143,6 +143,9 @@ public function isValid($value)
             try {
                 $value = new Varien_Simplexml_Element('<config>' . $value . '</config>');
             } catch (Exception $e) {
+                if (Mage::getIsDeveloperMode()) {
```

Comment:
This should be in a separate PR, lumping it here make it difficult to review because I'm not sure about the side-effects.